cpp/include/kvikio/posix_io.hpp

```diff
@@ -41,17 +41,8 @@ class StreamsByThread {
   std::map<std::pair<CUcontext, std::thread::id>, CUstream> _streams;
 
  public:
-  StreamsByThread() = default;
-  ~StreamsByThread() noexcept
-  {
-    for (auto& [_, stream] : _streams) {
-      try {
-        CUDA_DRIVER_TRY(cudaAPI::instance().StreamDestroy(stream));
-      } catch (const CUfileException& e) {
```

Comment:
it's interesting that this causes segfaults even with the catch.

Reply:
If I didn't miss anything, this is expected? I checked kvikio and cudf, and didn't find a part where we convert the `SIGSEGV` signal into a C++ exception. 

[This reference](https://en.cppreference.com/w/cpp/utility/program/signal) states that
> If the user defined function returns when handling SIGFPE, SIGILL, SIGSEGV or any other implementation-defined signal specifying a computational exception, the behavior is undefined.

This indicates the signal handler must cause the program to exit in a certain way (abnormally with `std::abort` or normally with `std::_Exit`). So I guess it's not quite possible (?) to handle the `SIGSEGV` "exception" after all without causing any UB problem.